packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2595,11 +2594,13 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
 
   bool _active = false;
 
+  /// Called when a hot-reload is triggered.
```

Comment:
I would take a look at the documentation on `State.reassemble`, which should probably be turned into a template for this.

Reply:
Is this really desired? This will make the IDE unable to get the documentation of `reassemble`

Reply:
How so?

Reply:
It makes the IDE (at least vscode) shows the macro code instead of its content. So we loose the ability to read the documentation from our IDE

Reply:
We should open an issue for that to track it.  Unfortunately, not using it for now means we have identically worded docs that get out of sync.  There are work arounds for this (going to the website or using tools that locally cache the website docs and make them available in the IDE).  I'm sure we can come up with something at the IDE level for it as well.

I would recommend making this member have the real documentation, and the implementer on `State` use it as a template.

Reply:
See https://github.com/flutter/flutter-intellij/issues/2524 for IntelliJ issue, https://github.com/Dart-Code/Dart-Code/issues/1420 for VSCode.  I think we can fix this.

Reply:
Sure, done 👌